### PR TITLE
Properly delete zip files

### PR DESCRIPTION
**Pull request by JSAbrahams**
See issue #229 

In addition to this, project files are now propely extracted. Before, projects contained a directory within which denoted the master branch which within contained the project. This is now directly extracted.

Before:
```
projects
 |- project0
  |- master
   |-files...
 |- project1
  |- master
   |-files...
```

After:
```
projects
 |- project0
  |-files...
 |- project1
  |-files...
```

### CODE REVIEW

File: modules/mining-pipeline/github-project-miner/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/miner/github/GitHubProjectDownloader.kt [hunk 1]
```diff
@@ -26,13 +26,18 @@ internal class GitHubProjectDownloader<P : Project>(
...
-    private fun File.getMasterFile(): File {
-        require(this.listFiles().isNotEmpty()) { "GitHub Project must contain files" }
-        return this.listFiles().first()
+    private fun File.extractMasterFile() {
+        require(this.listFiles().isNotEmpty()) { "GitHub Project must contain files." }
+        require(this.listFiles().first().isDirectory) { "GitHub Project must contain a directory for master." }
+
+        val masterFile = this.listFiles().first()
+        masterFile.listFiles()?.forEach { it.moveUpOneLevel() }
+        masterFile.delete()
```
Comment (FWDekker) [inline modules/mining-pipeline/github-project-miner/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/miner/github/GitHubProjectDownloader.kt]: Could you add an `assert` (or a `check` maybe?) that verifies that this method is called only on directories containing one subdirectory? It seems that this is an implicit assumption (which I think is reasonable), but I think we should be warned when this assumption is violated.

Comment (JSAbrahams) [reply modules/mining-pipeline/github-project-miner/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/miner/github/GitHubProjectDownloader.kt]: Check has been added. Before I checked that it wasn't empty, but now I explicitly check that there is only a single directory.

Comment (JSAbrahams) [reply modules/mining-pipeline/github-project-miner/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/miner/github/GitHubProjectDownloader.kt]: It was: `require(this.listFiles().isNotEmpty()) { "GitHub project must contain files." }`

Comment (FWDekker) [reply modules/mining-pipeline/github-project-miner/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/miner/github/GitHubProjectDownloader.kt]: > now I explicitly check that there is only a single directory

But doesn't the code you mentioned check only that there _are_ files, not that there is exactly one?

**Pull request merged**